cli/src/merge_tools/diff_working_copies.rs

```diff
@@ -99,10 +110,19 @@ fn check_out(
     tree: &MergedTree,
     sparse_patterns: Vec<RepoPathBuf>,
     options: &CheckoutOptions,
+    user_settings: &UserSettings,
 ) -> Result<TreeState, DiffCheckoutError> {
     std::fs::create_dir(&wc_dir).map_err(DiffCheckoutError::SetUpDir)?;
     std::fs::create_dir(&state_dir).map_err(DiffCheckoutError::SetUpDir)?;
-    let mut tree_state = TreeState::init(store, wc_dir, state_dir)?;
+    let tree_state_settings =
+        TreeStateSettings::try_from_user_settings(user_settings).map_err(|err| {
+            DiffCheckoutError::Other {
+                message: "failed to create the tree state settings from the user settings"
+                    .to_string(),
+                err: err.into(),
+            }
+        })?;
+    let mut tree_state = TreeState::init(store, wc_dir, state_dir, &tree_state_settings)?;
```

Comment:
Maybe this part isn't updated yet, but the external tool shouldn't enable EOL translation because the internal tools don't.

Reply:
Now, the external diff tool is updated to not enable EOL conversion, and added a test for that.